### PR TITLE
get posts by given category

### PR DESCRIPTION
**Pull request by astrocket**
Done

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -71,6 +71,13 @@ def author
     render_pages
   end
 
+  # GET /category/:category
```
Comment (sydneyitguy) [inline app/controllers/posts_controller.rb]: I think `/posts/tag/:tag` is better routes as `tag` is a column on posts.

File: config/routes.rb [hunk 2]
```diff
@@ -14,6 +14,8 @@
     end
   end
 
+  get '/category/:category', to: 'posts#category', constraints: { category: /([^\/]+?)(?=\.json|$|\/)/ }
```
Comment (sydneyitguy) [inline config/routes.rb]: 1. if we use `/posts/tag/:tag` route, this part can be included in `resources :posts`
2. we don't need this constraints here. we overrode the default value as `:author` can include dots (.), but `:tags` don't need that override

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (astrocket) [reply app/controllers/posts_controller.rb]: Checked.

File: config/routes.rb (hunk 2 above)
Comment (astrocket) [reply config/routes.rb]: I agree. I was thinking that the api route must match the front-end route. Will change to /posts/tag/:tag. Thanks for all comments ! 👍

**Pull request merged**